src/openforms/submissions/logic/actions.py

```diff
@@ -97,9 +148,56 @@ def apply(self, step: SubmissionStep, configuration: JSONObject) -> None:
             step._is_applicable = False
             step.data = DirtyData({})
 
+    def get_action_log_data(
+        self,
+        component_map: Dict[str, ComponentMeta],
+        all_variables: Dict[str, FormVariable],
+        initial_data: dict,
+    ) -> dict:
+        return {
+            "type_display": LogicActionTypes.get_choice(
+                LogicActionTypes.step_not_applicable
+            ).label,
+            "step_name": self.form_step_identifier,
+        }
+
+
+@dataclass
+class VariableAction(ActionOperation):
+    variable: str
+    value: JSONObject
+
+    @classmethod
+    def from_action(cls, action: ActionDict) -> "VariableAction":
+        return cls(variable=action["variable"], value=action["action"]["value"])
+
+    def get_action_log_data(
+        self,
+        component_map: Dict[str, ComponentMeta],
+        all_variables: Dict[str, FormVariable],
+        initial_data: dict,
+    ) -> dict:
+        # Check if it's a primitive value, which doesn't require introspection
+        if not isinstance(self.value, dict):
```

Comment:
in the previous patch I also saw some `isinstance` checks on `list` - is that still covered by `introspect_json_logic` then?

Reply:
Mhm I think actually it's a mistake here. Because the list may have json logic inside, so it should be introspected.